src/Cli/dotnet/Commands/Test/MTP/TestApplicationHandler.cs

```diff
@@ -85,6 +89,11 @@ private static string GetHandshakePropertyName(byte propertyId) =>
 
     internal void OnDiscoveredTestsReceived(DiscoveredTestMessages discoveredTestMessages)
     {
+        LogDiscoveredTests(discoveredTestMessages);
+
+        // TODO: If _handshakeInfo is null, we should error.
```

Comment:
will this TODO be fixed before this is merged? is there a tracking issue?
would it make sense to throw an InvalidOperationException if _handshakeInfo is null?

Reply:
@drognanar A different PR. An umbrella issue for this is https://github.com/dotnet/sdk/issues/50630